crypto/property/property.c

```diff
@@ -337,9 +337,8 @@ int ossl_method_store_fetch(OSSL_METHOD_STORE *store, int nid,
     int ret = 0;
     int j, best = -1, score, optional;
 
-#ifndef FIPS_MODULE
-    OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG, NULL);
-#endif
+    if (!OPENSSL_init_crypto(OPENSSL_INIT_LOAD_CONFIG, NULL))
+	return 0;
```

Comment:
The `#ifndef FIPS_MODULE` seems to be essential here. Without it, I get a linker error.
```
/usr/lib/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: providers/libfips.a(libfips-lib-property.o): in function `ossl_method_store_fetch':
/home/msp/src/openssl/crypto/property/property.c:340: undefined reference to `OPENSSL_init_crypto'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:23204: providers/fips.so] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/msp/src/openssl'
make: *** [Makefile:3208: build_sw] Error 2
```

Reply:
right.